python/ray/serve/_private/application_state.py

```diff
@@ -0,0 +1,217 @@
+import ray
+import traceback
+from typing import Dict, List
+from ray.serve._private.common import ApplicationStatus
+from ray.serve._private.deployment_state import DeploymentStateManager
+from ray.serve._private.common import (
+    DeploymentStatus,
+    DeploymentStatusInfo,
+    ApplicationStatusInfo,
+)
+import time
+from ray.exceptions import RayTaskError, RuntimeEnvSetupError
+import logging
+from ray.serve._private.constants import (
+    SERVE_LOGGER_NAME,
+)
+from ray.types import ObjectRef
+
+logger = logging.getLogger(SERVE_LOGGER_NAME)
+
+
+class ApplicationState:
+    """Manage single application states with all operations"""
+
+    def __init__(
+        self,
+        app_name: str,
+        deployment_state_manager: DeploymentStateManager,
+        deployment_params: List[Dict] = None,
+        deploy_obj_ref: ObjectRef = None,
+        deployment_time: float = 0,
+    ):
+        """
+        Args:
+            app_name: application name
+            deployment_state_manager: deployment state manager which is used for
+                fetching deployment information
+            deployment_params: all deployment parameters to deploy.
+            deploy_obj_ref: Task ObjRef of deploying application.
+            deployment_time: Deployment timestamp
+        """
+        if deploy_obj_ref:
+            self.status: ApplicationStatus = ApplicationStatus.DEPLOYING
+        else:
+            self.status: ApplicationStatus = ApplicationStatus.NOT_STARTED
+        self.app_name = app_name
+        self.deployment_params = deployment_params
+        self.to_be_deleted = False
```

Comment:
```suggestion
        self.all_deployments_deleted = False
```

Reply:
this attribute is under ApplicationState to indicate whether this application should be deleted or not. In the ApplicationStateManager update(), I think the attribute name is clearer to indicate "the application can be deleted" instead of "all deployments have been deleted". WDYT?